jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlConfiguration.java

```diff
@@ -1734,47 +1756,64 @@ Args asEmptyVarArgs(Class<?> varArgType)
-                    String name = names.get(p);
-                    if (name != null)
+                    // If we don't have any parameters with names, then no match
+                    Annotation[][] parameterAnnotations = executable.getParameterAnnotations();
+                    if (parameterAnnotations == null || parameterAnnotations.length == 0)
+                        return null;
+
+                    // Find the position of all named parameters from the executable
+                    Map<String, Integer> position = new HashMap<>();
+                    int p = 0;
+                    for (Annotation[] paramAnnotation : parameterAnnotations)
                     {
-                        Integer pos = position.get(name);
-                        if (pos == null)
-                            return null;
-                        if (pos != p)
+                        Integer pos = p++;
+                        Arrays.stream(paramAnnotation)
+                            .filter(Name.class::isInstance)
+                            .map(Name.class::cast)
+                            .findFirst().ifPresent(n -> position.put(n.value(), pos));
+                    }
+
+                    List<Object> arguments = new ArrayList<>(_arguments);
+                    List<String> names = new ArrayList<>(_names);
+                    // Map the actual arguments to the names
+                    for (p = 0; p < count; p++)
+                    {
+                        String name = names.get(p);
+                        if (name != null)
                         {
-                            // adjust position of parameter
-                            arguments.add(pos, arguments.remove(p));
-                            names.add(pos, names.remove(p));
-                            p = Math.min(p, pos);
+                            Integer pos = position.get(name);
+                            if (pos == null)
+                                return null;
+                            if (pos != p)
+                            {
+                                // adjust position of parameter
+                                arguments.add(pos, arguments.remove(p));
+                                names.add(pos, names.remove(p));
+                                p = Math.min(p, pos);
+                            }
                         }
                     }
+                    args = arguments.toArray(new Object[0]);
                 }
-                return arguments.toArray(new Object[0]);
+
+                // Check assignable
+                Parameter[] params = executable.getParameters();
+                for (int i = 0; i < args.length; i++)
+                {
+                    if (args[i] != null &&
+                        !params[i].getType().isAssignableFrom(args[i].getClass()) &&
+                        !TypeUtil.isUnboxable(params[i].getType(), args[i]))
+                        return null;
+                }
```

Comment:
I'm not sure I understand this.
We want to return `args` if unboxing is possible? That is already done by reflection.
Removing this block of code does not fail any test, so I am not sure why it was added?